devops/openshift/api-deploy.yml

```diff
@@ -245,6 +248,18 @@ objects:
       to:
         kind: Service
         name: ${NAME}
+  - apiVersion: autoscaling/v1
+    kind: HorizontalPodAutoscaler
+    metadata:
+      name: ${NAME}-hpa
+    spec:
+      scaleTargetRef:
+        apiVersion: apps/v1
+        kind: ReplicationController
+        name: ${NAME}-rc
+      minReplicas:  "${{REPLICAS}}"
+      maxReplicas: 10
```

Comment:
Does it take into consideration the available quota and stop creating more replicas?
I noticed that all PODs are configured to a max of 10, just wondering if we need to plan the maxReplicas as per the POD application type.

Reply:
We cannot create pods more than the available quota, it will not create more replicas if the resource is not available in openshift. During that situation, sysdig alerts come into place, that alerts the users that the overloading is happening and the cluster needs some attention. I mentioned the max replicas as 10, because depending on the load, different containers will scaleup to different replica sets, and it will come down once the utilization is less than 80%. I believe at a point not all the containers will go to replicate 10. Hopefully not

Reply:
We cannot create pods more than the available quota, it will not create more replicas if the resource is not available in openshift. During that situation, sysdig alerts come into place, that alerts the users that the overloading is happening and the cluster needs some attention. I mentioned the max replicas as 10, because depending on the load, different containers will scaleup to different replica sets, and it will come down once the utilization is less than 80%. I believe at a point not all the containers will go to replicate 10. Hopefully not